tests/csrf_tests/tests.py

```diff
@@ -294,6 +294,19 @@ def test_https_bad_referer(self):
             status_code=403,
         )
 
+    @override_settings(ALLOWED_HOSTS=['www.example.com'])
+    def test_https_malformed_host(self):
+        """
+        The bad request page loads if the host header is incorrect.
+        """
+        req = self._get_GET_no_csrf_cookie_request()
+        req._is_secure_override = True
+        req.META['HTTP_HOST'] = '@malformed'
+        req.META['HTTP_REFERER'] = 'https://www.evil.org/somepage'
+        req.META['SERVER_PORT'] = '443'
+        resp = requires_csrf_token(token_view)(req)
+        self.assertEqual(resp.status_code, 200)
```

Comment:
I'm not sure I totally understand this test.

Does it actually involve the bad-request page? It doesn't look like it, but the docstring says so.

Why is the status code 200, rather than 400? This looks to me like a request with an `HTTP_HOST` that doesn't match `ALLOWED_HOSTS`.

Reply:
No, it doesn't. `requires_csrf_token(token_view)` is a standin for the default 400 handler, but it gives status 200. I will improve the docstring.

Reply:
Much clearer, thanks!